backend/data/data.go

```
@@ -119,6 +125,131 @@ func (data *api) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+		controllableUnitBusinessID = *cuLookupRequestBody.ControllableUnitBusinessID
+	}
+
+	accountingPointID := ""
+	if cuLookupRequestBody.AccountingPointID != nil {
+		accountingPointID = *cuLookupRequestBody.AccountingPointID
+	}
+
+	if endUserBusinessID == "" {
+		writeErrorToResponseWriter(w, http.StatusBadRequest, errorMessage{ //nolint:exhaustruct
+			Message: "missing end user business ID",
+		})
+		return
+	}
+
+	if accountingPointID == "" && controllableUnitBusinessID == "" {
+		writeErrorToResponseWriter(w, http.StatusBadRequest, errorMessage{ //nolint:exhaustruct
+			Message: "missing accounting point ID or business ID",
+		})
+		return
+	}
+
+	slog.InfoContext(
+		ctx, "will lookup controllable unit",
+		"end_user_business_id", endUserBusinessID,
+		"accounting_point_id", accountingPointID,
+		"controllable_unit_business_id", controllableUnitBusinessID,
+	)
+
+	conn, err := data.db.Acquire(ctx)
+	if err != nil {
+		slog.ErrorContext(ctx, "could not acquire system connection", "error", err)
+		writeErrorToResponseWriter(w, http.StatusInternalServerError, errorMessage{ //nolint:exhaustruct
+			Message: "could not acquire system connection",
+		})
+		return
+	}
+	defer conn.Release()
+	tx, err := conn.Begin(ctx)
+	if err != nil {
+		slog.ErrorContext(ctx, "could not start transaction", "error", err)
+		writeErrorToResponseWriter(w, http.StatusInternalServerError, errorMessage{ //nolint:exhaustruct
+			Message: "could not start transaction",
+		})
+		return
+	}
+	queries := models.New(tx)
+
+	cuLookup, err := queries.ControllableUnitLookup(
+		ctx,
+		endUserBusinessID,
+		controllableUnitBusinessID,
+		accountingPointID,
+	)
+	if err != nil {
+		slog.ErrorContext(ctx, "CU lookup query failed", "error", err)
+	}
+	if err != nil || len(cuLookup) == 0 {
+		writeErrorToResponseWriter(w, http.StatusNotFound, errorMessage{ //nolint:exhaustruct
+			Message: "controllable unit not found",
+		})
+		return
+	}
+	if err = tx.Commit(ctx); err != nil {
```

shouldn't we just defer and ignore error-checking? There is basically nothing to commit.

---

There are events to commit now.

---

Ah good point. Shouldn't we return a HTTP 500 if we are not able to commit the events then?